CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Review/ReviewViewController.swift

```diff
@@ -2,302 +2,433 @@
+        addPagesButton.configureButton(image: UIImageNamedPreferred(named: "plus_icon") ?? UIImage(),
+                                       name:
+                        NSLocalizedStringPreferredFormat("ginicapture.multipagereview.secondaryButtonTitle",
+                                                        comment: "Add pages button title"))
+        addPagesButton.isHidden = !giniConfiguration.multipageEnabled
+        addPagesButton.didTapButton = { [weak self] in
+            guard let self = self else { return }
+            self.setCellStatus(for: self.currentPage, isActive: false)
+            self.delegate?.reviewDidTapAddImage(self)
+        }
+        return addPagesButton
+    }()
+
+    private lazy var cellSize: CGSize = {
+        return calculatedCellSize()
     }()
-    
-    // Properties
-    fileprivate var imageViewBottomConstraint: NSLayoutConstraint!
-    fileprivate var imageViewLeadingConstraint: NSLayoutConstraint!
-    fileprivate var imageViewTopConstraint: NSLayoutConstraint!
-    fileprivate var imageViewTrailingConstraint: NSLayoutConstraint!
-    fileprivate var currentDocument: GiniCaptureDocument?
-    fileprivate var giniConfiguration: GiniConfiguration
-    
-    // Images
-    fileprivate var rotateButtonImage: UIImage? {
-        return UIImageNamedPreferred(named: "reviewRotateButton")
+
+    private var currentPage: Int = 0 {
+        didSet {
+            setCellStatus(for: currentPage, isActive: true)
+        }
     }
-    
-    /**
-     Designated initializer for ReviewViewController
-     
-     - parameter document: Document to be reviewed
-     - paramter giniConfiguration: `GiniConfiguration`
-     
-     - note: Component API only.
-     */
-    public init(document: GiniCaptureDocument, giniConfiguration: GiniConfiguration) {
+
+    // This is needed in order to "catch" the screen rotation on the modally presented viewcontroller
+    private var previousScreenHeight: CGFloat = 0
+
+    // MARK: - Init
+
+    public init(pages: [GiniCapturePage], giniConfiguration: GiniConfiguration) {
+        self.pages = pages
         self.giniConfiguration = giniConfiguration
-        self.currentDocument = document
         super.init(nibName: nil, bundle: nil)
     }
-    
-    /**
-     Returns an object initialized from data in a given unarchiver.
-     
-     - warning: Not implemented.
-     */
-    public required init?(coder aDecoder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
+
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(imageDocuments:) has not been implemented")
     }
-    
-    public override func loadView() {
-        super.loadView()
+}
+
+// MARK: - UIViewController
+
+extension ReviewViewController {
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        title = NSLocalizedStringPreferredFormat("ginicapture.multipagereview.title",
```

Comment:
[Suggestion] Could you please extract all view setup-related lines from viewDidLoad to `setupView()` method?